Model/Order/PlaceOrder.php

```diff
@@ -592,9 +414,14 @@ private function buildOrderData(array $firstTransaction, int $quoteId, string $p
         );
         $orderPayment->setIsTransactionClosed(true); // @phpstan-ignore method.notFound
 
+
+        /** @var TransactionInterface $transaction */
+        $transaction = $this->transactionFactory->create();
         $transaction->setTxnId($firstTransaction['transaction_id']);
         $transaction->setTxnType(TransactionInterface::TYPE_PAYMENT); // TODO: verify this transaction type is correct
 
+        /** @var OrderDataInterface $orderData */
+        $orderData = $this->orderDataFactory->create();
```

Comment:
[Preference]: I prefer to have variables be initialized right before they are used, but not sure if we have a coding standard around this.

Reply:
My personal preference is to keep all variable initializations together so that they're easier to find.